src/auth/src/credentials/mds_credential.rs

```diff
@@ -12,38 +12,154 @@
+//! let token = credential.get_token().await?;
+//! println!("Token: {}", token.token);
+//! # Ok::<(), CredentialError>(())
+//! # });
+//! ```
+//!
+//! [Cloud Run]: https://cloud.google.com/run
+//! [default service account]: https://cloud.google.com/iam/docs/service-account-types#default
+//! [gce-link]: https://cloud.google.com/products/compute
+//! [gke-link]: https://cloud.google.com/kubernetes-engine
+//! [Metadata Service]: https://cloud.google.com/compute/docs/metadata/overview
+
 use crate::credentials::dynamic::CredentialTrait;
-use crate::credentials::{Credential, Result};
+use crate::credentials::{Credential, Result, DEFAULT_UNIVERSE_DOMAIN, QUOTA_PROJECT_KEY};
 use crate::errors::{is_retryable, CredentialError};
 use crate::token::{Token, TokenProvider};
 use async_trait::async_trait;
 use bon::Builder;
 use http::header::{HeaderName, HeaderValue, AUTHORIZATION};
 use reqwest::Client;
+use std::default::Default;
 use std::sync::Arc;
 use std::time::Duration;
 
 const METADATA_FLAVOR_VALUE: &str = "Google";
 const METADATA_FLAVOR: &str = "metadata-flavor";
-const METADATA_ROOT: &str = "http://metadata.google.internal/computeMetadata/v1";
+const METADATA_ROOT: &str = "http://metadata.google.internal/";
 
 pub(crate) fn new() -> Credential {
-    let token_provider = MDSAccessTokenProvider::builder()
-        .endpoint(METADATA_ROOT)
-        .build();
-    Credential {
-        inner: Arc::new(MDSCredential { token_provider }),
-    }
+    MDSCredentialBuilder::default().build()
 }
 
 #[derive(Debug)]
 struct MDSCredential<T>
 where
     T: TokenProvider,
 {
+    quota_project_id: Option<String>,
+    universe_domain: Option<String>,
     token_provider: T,
 }
 
+/// Creates [Credential] instances backed by the [Metadata Service].
+///
+/// While the Google Cloud client libraries for Rust default to credentials
+/// backed by the metadata service, some applications may need to:
+/// * Customize the metadata service credentials in some way
+/// * Bypass the [Application Default Credentials] lookup and only
+///    use the metadata server credentials
+/// * Use the credentials directly outside the client libraries
+///
+/// [Application Default Credentials]: https://cloud.google.com/docs/authentication/application-default-credentials
+/// [Metadata Service]: https://cloud.google.com/compute/docs/metadata/overview
+#[derive(Debug, Default)]
+pub struct MDSCredentialBuilder {
+    endpoint: Option<String>,
+    quota_project_id: Option<String>,
+    scopes: Option<Vec<String>>,
+    universe_domain: Option<String>,
+}
+
+impl MDSCredentialBuilder {
+    /// Sets the endpoint for this credential.
+    /// If not set, the credentials use `http://metadata.google.internal/`.
+    pub fn endpoint<S: Into<String>>(mut self, endpoint: S) -> Self {
+        self.endpoint = Some(endpoint.into());
+        self
+    }
+
+    /// Set the [quota project] for this credential.
+    /// In some services, you can use a service account in
```

Comment:
nit: here and everywhere else, separate short description from detailed description:

https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components

```suggestion
    /// Set the [quota project] for this credential.
    ///
    /// In some services, you can use a service account in
```